docs/ProcessingPipeline.md

```diff
@@ -0,0 +1,62 @@
+# Processing Pipeline
+
+Some applications might require that traces be altered or filtered out before they are sent upstream. The processing pipeline allows users to create *processors* to define such behavior.
+
+Processors can be any object that responds to `#call` accepting `trace` as an argument (which is an `Array` of `Datadog::Span`s.)
+
+For example:
+
+```ruby
+lambda_processor = ->(trace) do
+  # Processing logic...
+  trace
+end
+
+class MyCustomProcessor
+  def call(trace)
+    # Processing logic...
+    trace
+  end
+end
+custom_processor = MyFancyProcessor.new
+```
+
+`#call` blocks of processors *must* return the `trace` object; this return value will be passed to the next processor in the pipeline.
+
+These processors must then be added to the pipeline via `Datadog::Pipeline.before_flush`:
+
+```ruby
+Datadog::Pipeline.before_flush(lambda_processor, custom_processor)
+```
+
+You can also define processors using the short-hand block syntax for `Datadog::Pipeline.before_flush`:
+
+```ruby
+Datadog::Pipeline.before_flush do |trace|
+  trace.delete_if { |span| span.name =~ /forbidden/ }
+end
+```
+
+## Filtering
+
+You can use the `Datadog::Pipeline::SpanFilter` processor to remove spans, when the block evaluates as truthy:
+
+```ruby
+Datadog::Pipeline.before_flush(
+  # Remove spans that match a particular resource
+  Datadog::Pipeline::SpanFilter.new { |span| span.resource =~ /PingController/ },
+  # Remove spans that are trafficked to localhost
+  Datadog::Pipeline::SpanFilter.new { |span| span.get_tag('host') == 'localhost' }
+)
+```
+
+## Processing
+
+You can use the `Datadog::Pipeline::SpanProcessor` processor to modify spans:
+
+```ruby
+Datadog::Pipeline.before_flush(
+  # Strip matching text from the resource field
+  Datadog::Pipeline::SpanProcessor.new { |span| span.resource.gsub!(/password=.*/, '') }
+)
+```
```

Comment:
Is it just me or the order of presentation in these docs is a bit... funky? E.g. we start with the most complex/configurable (a fully custom processor), and then after explaining all that we come in with "ah btw we have shorthands".

Perhaps we could instead structure this section as:

```
└── Processing pipeline
    ├── Intro to what it is
    ├── Filtering spans
    ├── Modifying spans
    └── Advanced trace modifications
```

Reply:
Agreed! 👍